common-app-lib/src/cmr/common_app/services/kms_fetcher.clj

```diff
@@ -60,36 +62,50 @@
+          _ (rl-util/log-redis-reading-start "kms-fetcher.fetch-gcmd-keywords-map" kms-cache-key)
+          [t1 kms-cache-value] (util/time-execution (cache/get-value kms-cache kms-cache-key))]
+      (rl-util/log-redis-read-complete "fetch-gcmd-keywords-map" kms-cache-key t1)
+      (kms-lookup/create-kms-index
+        context
+        (into {}
+              (for [keyword-scheme (keys kms/keyword-scheme->field-names)]
+                ;; if the keyword-scheme-value is nil that means we could not get the KMS keywords
+                ;; in this case use the cached value instead so that we don't wipe out the cache.
+                (if-let [keyword-scheme-value (kms/get-keywords-for-keyword-scheme context keyword-scheme)]
+                  [keyword-scheme keyword-scheme-value]
+                  [keyword-scheme (get kms-cache-value keyword-scheme)])))))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "fetch-gcmd-keywords-map found redis carmine exception. Will return nil result." e)
+          nil)
+        (throw e)))))
 
 (defn get-kms-index
   "Retrieves the GCMD keywords map from the cache."
   [context]
-  (when-not (:ignore-kms-keywords context)
-    (let [cache (cache/context->cache context kms-cache-key)
-          _ (rl-util/log-redis-reading-start kms-cache-key)
-          [t1 kms-index] (util/time-execution (cache/get-value cache kms-cache-key))]
-      (if kms-index 
-        (do 
-          (rl-util/log-redis-read-complete "get-kms-index" kms-cache-key t1)
-          kms-index)
-        (let [[t1 kms-index] (util/time-execution (fetch-gcmd-keywords-map context))
-              [t2 _] (util/time-execution (cache/set-value cache kms-cache-key kms-index))]
-          (rl-util/log-data-gathering-stats "get-kms-index" kms-cache-key t1)
-          (rl-util/log-redis-write-complete "get-kms-index" kms-cache-key t2)
-          kms-index)))))
+  (try
+    (when-not (:ignore-kms-keywords context)
+      (let [cache (cache/context->cache context kms-cache-key)
+            _ (rl-util/log-redis-reading-start kms-cache-key)
+            [t1 kms-index] (util/time-execution (cache/get-value cache kms-cache-key))]
+        (if kms-index
+          (do
+            (rl-util/log-redis-read-complete "get-kms-index" kms-cache-key t1)
+            kms-index)
+          (let [[t1 kms-index] (util/time-execution (fetch-gcmd-keywords-map context))
+                [t2 _] (util/time-execution (cache/set-value cache kms-cache-key kms-index))]
+            (rl-util/log-data-gathering-stats "get-kms-index" kms-cache-key t1)
+            (rl-util/log-redis-write-complete "get-kms-index" kms-cache-key t2)
+            kms-index))))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "get-kms-index found redis carmine exception. Will return nil result." e)
+          nil)
```

Comment:
Remove the (do and the nil)
(error returns nil.

Please do this in the kms_lookup namespace as well.